src/types/contexts/user.ts

```diff
@@ -0,0 +1,25 @@
+export interface UserRoles {
+  hall: string[];
+  staff: string[];
+  ashc: string[];
+}
+
+export interface User {
+  uid: string;
+  email: string;
+  emailVerified?: boolean;
+  displayName: string;
+  hall: string;
+  hallVerified: boolean;
+  roomNumber: number;
```

Comment:
this is optional, correct?
```suggestion
  roomNumber?: number;
```

Reply:
also i'd like to reiterate that I'm not sure every hall uses only numbers for room identifiers

Reply:
4b125d4332995988b0d364172dcae439f7d109b5